servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ClientStrategyInfluencerChainBuilder.java

```diff
@@ -46,32 +58,68 @@ private ClientStrategyInfluencerChainBuilder(ClientStrategyInfluencerChainBuilde
     }
 
     void add(StreamingHttpClientFilterFactory clientFilter) {
-        clientChain = clientChain.merge(clientFilter.requiredOffloads());
+        HttpExecutionStrategy filterOffloads = clientFilter.requiredOffloads();
+        if (defaultStrategy() == filterOffloads || noOffloadsStrategy() == filterOffloads) {
```

Comment:
It is really late in the game but I am tempted to redefine `noOffloadsStrategy` so that it is consistent with what you suggest and define another "neverOffloadStrategy()" constant for the "force no offloading". I am concerned that the inconsistent behavior between `noOffloadsStrategy` for builder (which means **disable offloading**) and filters (which means "no offloading needed").

Reply:
Sounds reasonable. Only one worry: what will happen with 0.41 users who already use `noOffloadsStrategy()` on the builder after they upgrade to 0.42? Will that work as in 0.41? If not, we can consider doing the following:
1. Deprecate `noOffloadsStrategy()` and introduce `neverOffloadStrategy()` as a replacement they should use.
2. Rename `anyStrategy()` to `noOffloadingStrategy()` (slightly different name just to avoid changing behavior for the pre-existing name)
More hustle for users to rename what they use, but should keep the expected behavior IIUC. WDYT?

Reply:
OK, I will propose this as a separate PR.

Reply:
> So, `defaultStrategy()` can be replaced with `offloadAll()`

This is problematic because the API conversion step applies different offloads for the default and converting default to offloadAll at the factory addition stage would force additional unnecessary offloading. I am still tempted to have using default be an error.

Reply:
The `DefaultHttpExecutionStrategyInfluencer` default impl is `offloadAll()`. If users override it and return `defaultStrategy()` instead, does it change anything? IIRC, filter chain computation is always additive.

Reply:
Returning `defaultStrategy()`probably doesn't have any effect except that is more ambiguous than `offloadAll()` and if we substitute `offloadAll()` we lose the ability of API conversions to substitute less offloading when default is presented. I'd prefer if influencers only returned non-special execution strategies and I don't see any useful situation for them to return special strategies.

Reply:
`defaultStrategy()` is now allowed without generating a warning. I think the docs should still suggest that it is not good to return it from `requiredOffloads()`